Trash Bin

BrodBlox09 opened the pull request:
Add functionality for trash bin. Graphics still need to be added. Currently, placeholders are in use.

BrodBlox09 committed:
Add placeholder graphics for trashbin

BrodBlox09 committed:
Add trashbin functions

BrodBlox09 committed:
Remove corrupted image file

BrodBlox09 committed:
Rename trashbin to trash_bin

BrodBlox09 committed:
Add license headers to script files

BrodBlox09 committed:
Update TSC to no longer remove comments on compilation

To adhere to the requirement of a copyright header at the top of each file.

BrodBlox09 committed:
Add license headers to files

BrodBlox09 committed:
Update copyright headers

BrodBlox09 committed:
Add frying pan & pickup ability

BrodBlox09 committed:
Allow players in adventure mode to place frying pan

BrodBlox09 committed:
Merge branch 'main' into trashbin

BrodBlox09 committed:
Specify frying pan interaction

BrodBlox09 committed:
Merge branch 'trashbin' of https://github.com/BrodBlox09/Cook-Off into trashbin

BrodBlox09 commented:
Resolved merge conflict.

BrodBlox09 commented:
Added the basic display of the frying pan with placeholders, to be replaced once models and graphics are provided.

BrodBlox09 committed:
Delete extraneous entity definition

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/BedrockCommands/Cook-Off?pullRequest=5) <br/>All committers have signed the CLA.

BrodBlox09 committed:
Enable comments in TypeScript configuration

Undo remove line 7

jeanmajid commented on behavior_pack/src/PlayerInventory.ts:
```diff
@@ -0,0 +1,29 @@
+	}
+
+	public give(itemStack: ItemStack): void {
+		let slotIndex = this.container.firstEmptySlot();
```
Make sure you check if a empty slot even exists

BrodBlox09 replied:
Resolved in commit 8db13ded455a2878ce7997b05172046cbd631f64

jeanmajid commented on behavior_pack/src/Utils.ts:
```diff
@@ -0,0 +1,21 @@
+
+export default class Utils {
+	static getOverworld(): Dimension {
+		return world.getDimension("minecraft:overworld");
```
cache the dimension, calling getDimension is expensive

BrodBlox09 replied:
Resolved in commit 4c8a8759412fb18413225bcc28c47c48c9288802

jeanmajid commented on behavior_pack/src/index.ts:
```diff
@@ -1,5 +1,30 @@
+			}
+		}
+	}
+});
```
remove this whole runInterval for now, wait until the counter is implemented

BrodBlox09 replied:
Removed in commit ba57d1f1d6c7864bc5e7fc414c2981abe3d93b67

jeanmajid reviewed (changes requested):
Make sure you Use good practises like const and === instead of let and == wherever possible. The code will have to get restructured into folders, but this is already not bad

BrodBlox09 committed:
Update method for placing blocks in adventure

Instead of looping through every player's inventory's slots every tick to ensure all blocks that are supposed to be allowed to be placed in adventure mode are placeable, make them placeable when the blocks are given to the player.

BrodBlox09 committed:
Update frying pan

Make frying pan placeable in adventure mode with new system.

BrodBlox09 committed:
Make give function more robust

Now acts more like the /give command.

BrodBlox09 committed:
Follow programming best practices

Replace == and let with === and const when applicable.

BrodBlox09 committed:
Update Vector.ts

Make the file more concise.

BrodBlox09 committed:
Cache overworld dimension

BrodBlox09 committed:
Define return type of Utils.permutationWithState()

Return type is now BlockPermutation

BrodBlox09 committed:
Specialize frying pan behavior

Frying pan now uses its own custom component.

BrodBlox09 committed:
TSC
